### PR TITLE
drop support for Python 3.8

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

Numpy is imminently dropping support for Python 3.8

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by bmorris3** [comment]
🚨 For those curious, "imminent" means [_today_](https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table). 🚨

**Comment by zacharyburnett** [comment]
The `Required Status Checks` portion of the branch protections will also need to be updated after merge (`Windows - Python 3.8` -> `Windows - Python 3.9`

**Comment by pllim** [comment]
Gotta also undo https://github.com/spacetelescope/jdaviz/pull/2173

**Comment by pllim** [comment]
Re: https://github.com/spacetelescope/jdaviz/pull/2152#issuecomment-1508679245

That is correct, so only admin can merge this one because it requires branch protection override.

**Review by rosteen — APPROVED** [review]
Thanks, I'll change the branch protection and drop the `ipython` pin after merge.

File: setup.cfg
Comment (kecnry) [inline setup.cfg:18]: If a user has python 3.8 and installs or attempts to upgrade jdaviz via pip, will they then just get the most recent release _before this PR_?  If so, I'm concerned that this might just result in users not getting bugfixes and new features (either intentionally because they don't want to update their python environment or just because they don't realize they need to update in order to get newer releases).

If our minimum requirements on upstream packages do not require 3.9 (yet), then would we be better off holding off as long as possible?

Comment (rosteen) [reply setup.cfg:18]: I understand the concern, but with both `numpy` and `astropy` already requiring 3.9, these users already wouldn't be getting new features in relevant packages (I assume astropy LTS still supports 3.8 for bug fixes at least). We can make the python requirement change extra prominent in the 3.5 release notes, but I think it's time for this. Users stuck on 3.8 would still get bug fix releases for 3.4, but also I'm skeptical that users who haven't upgraded from 3.8 are upgrading their packages either.

**Comment by rosteen** [comment]
Merging, I'll also add a warning/note in the docs about this when I open the PR to change the `ipython` pin.

**Pull request merged**
